include/fiction/algorithms/physical_design/post_layout_optimization.hpp

```diff
@@ -918,65 +364,92 @@ class post_layout_optimization_impl
+        timeout_limit_reached        = false;
 
         if (max_gate_relocations == 0)
         {
-            fiction::wiring_reduction_stats wiring_reduction_stats{};
-            fiction::wiring_reduction(layout, &wiring_reduction_stats);
+            // perform wiring reduction only when no gate relocations are allowed
+            fiction::wiring_reduction_stats  wiring_reduction_stats{};
+            fiction::wiring_reduction_params wiring_reduction_params{};
+
+            // update the remaining timeout
+            wiring_reduction_params.timeout = update_timeout();
+
+            if (!timeout_limit_reached)
+            {
+                fiction::wiring_reduction(layout, wiring_reduction_params, &wiring_reduction_stats);
+            }
         }
         else
         {
-            while (moved_at_least_one_gate || reduced_wiring)
+            // iteratively optimize the layout until no more improvements can be made or timeout is reached
+            while ((moved_at_least_one_gate || reduced_wiring) && !timeout_limit_reached)
             {
                 reduced_wiring = false;
-                fiction::wiring_reduction_stats wiring_reduction_stats{};
-                if (moved_at_least_one_gate && !ps.optimize_pos_only)
+
+                // update the remaining timeout
+                fiction::wiring_reduction_params wiring_reduction_params{};
+                wiring_reduction_params.timeout = update_timeout();
+
+                if (moved_at_least_one_gate && !ps.optimize_pos_only && !timeout_limit_reached)
                 {
-                    fiction::wiring_reduction(layout, &wiring_reduction_stats);
-                    if ((wiring_reduction_stats.area_improvement != 0ull) ||
-                        (wiring_reduction_stats.wiring_improvement != 0ull))
+                    fiction::wiring_reduction_stats wiring_reduction_stats{};
+                    fiction::wiring_reduction(layout, wiring_reduction_params, &wiring_reduction_stats);
+
+                    // check if wiring reduction made any improvements
+                    if (wiring_reduction_stats.area_improvement != 0ull ||
+                        wiring_reduction_stats.wiring_improvement != 0ull)
                     {
                         reduced_wiring = true;
                     }
                 }
 
-                std::vector<tile<Lyt>> gate_tiles{};
+                // gather all relevant gate tiles for relocation
+                std::vector<tile<Lyt>> gate_tiles;
                 gate_tiles.reserve(layout.num_gates() + layout.num_pis() - layout.num_pos());
                 layout.foreach_node(
                     [&layout, &gate_tiles](const auto& node)
                     {
-                        if (const tile<Lyt> tile = layout.get_tile(node);
-                            (layout.is_gate(node) && !layout.is_wire(node)) || layout.is_fanout(node) ||
+                        const tile<Lyt> tile = layout.get_tile(node);
```

Comment:
It doesn't make a difference here, but it is generally preferred to move the condition variable declaration inside the `if` statement condition to limit its scope. This leads to less pollution and faster resource deallocation.